From 9d9effca9d7d7cf6341182a7c5cabcbd6fa28063 Mon Sep 17 00:00:00 2001
From: Xin Long <lucien.xin@gmail.com>
Date: Fri, 21 Oct 2022 10:22:47 -0400
Subject: [PATCH] ethtool: eeprom: fix null-deref on genl_info in dump
Git-commit: 9d9effca9d7d7cf6341182a7c5cabcbd6fa28063
Patch-mainline: v6.1-rc3
References: git-fixes
Modified-by-SEL: Yes, refreshed due to context change


The similar fix as commit 46cdedf2a0fa ("ethtool: pse-pd: fix null-deref on
genl_info in dump") is also needed for ethtool eeprom.

Fixes: c781ff12a2f3 ("ethtool: Allow network drivers to dump arbitrary EEPROM data")
Signed-off-by: Xin Long <lucien.xin@gmail.com>
Link: https://lore.kernel.org/r/5575919a2efc74cd9ad64021880afc3805c54166.1666362167.git.lucien.xin@gmail.com
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Signed-off-by: Jun Yang <jun.yang@suse.com>
---
 net/ethtool/eeprom.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/ethtool/eeprom.c
+++ b/net/ethtool/eeprom.c
@@ -52,7 +52,7 @@ static int eeprom_prepare_data(const str
 		goto err_free;
 
 	ret = dev->ethtool_ops->get_module_eeprom_by_page(dev, &page_data,
-							  info->extack);
+							  info ? info->extack : NULL);
 	if (ret < 0)
 		goto err_ops;
 
